### ISSUE TITLE
Perform WCAG 2.0 "AA" Assessment

### ISSUE THREAD

**Issue by mweech** [fgpv-vpgf/fgpv-vpgf#1074]
Perform WCAG 2.0 AA Assessment on Viewer and attach results to this issue.

**Comment by dan-bowerman** [comment]
# 1.1 Text Alternatives
## 1.1.1 Non-text content
* [ ] FAIL. RAMP2 mostly handles this quite well, virtually all non-text elements reveal their purpose upon highlighting. 

### Grey areas:
* The map and overlaid data cannot appropriately be described with text descriptions -- the data grid is the best chance of achieving this.
* In RAMP1, a simplified grid displayed the data in the current map extent, RAMP2 does not have this functionality. In order to at least meet the level of accessibility in RAMP1, an extent-based view of the data should be available.

### Actions for WCAG "AA" compliance:
* [ ] The map should describe itself as such with aria-tagging, if possible. Since the map cannot be described accurately, just describing the element as an interactive map object should be enough.
* [ ] The mini-map should similarly be describing its purpose.
* [ ] This is a nitpick, and not an accessibility concern, but the mini-map 'collapse' arrow should use the same hover text styling as the other non-text elements. 

***

#1.2 Time-based Media
## 1.2.1 Audio-only and Video-only (Prerecorded)
* [x] PASS. Not an issue for RAMP as of August 2016. May be an issue for custom deployments of RAMP (e.g. the "Peter Kent announcement" video from JOSM's web map)

## 1.2.2 Captions (Prerecorded)
* [x] PASS. Not an issue for RAMP as of August 2016.

## 1.2.3 Audio Description or Media Alternative (Prerecorded)
* [x] PASS. Not an issue for RAMP as of August 2016.

## 1.2.4 Captions (Live)
* [x] PASS. Not an issue for RAMP as of August 2016.

## 1.2.5 Audio Description (Prerecorded)
* [x] PASS. Not an issue for RAMP as of August 2016.

***

# 1.3 Adaptable
## 1.3.1 Info and Relationships
TBD
## 1.3.2 Meaningful Sequence
TBD
## 1.3.3 Sensory Characteristics
TBD

***

# 1.4 Distinguishable
## 1.4.1 Use of Color
TBD
## 1.4.2 Audio Control
* [x] PASS. Not an issue for RAMP as of August 2016.
## 1.4.3 Contrast (Minimum)
TBD
## 1.4.4 Resize text
* [x] PASS. RAMP handles this very well.
## 1.4.5 Images of Text
* [x] PASS. Not an issue for RAMP as of August 2016, though this could change with custom deployments that leverage the logo in the top left corner of the 
***

# 2.1 Keyboard Accessible
## 2.1.1 Keyboard
TBD
## 2.1.2 No Keyboard Trap
* [ ] FAIL. Currently there is a bug where displaying !(https://files.gitter.im/fgpv-vpgf/NztJ/blob) on tabbing through the application traps the keyboard and seemingly provides no additional functionality at this time. 
***

# 2.2 Enough Time
## 2.2.1 Timing Adjustable
* [x] PASS. Not an issue for RAMP as of August 2016. Care for this criteria will need to be paid if animation / time series support is implemented.
## 2.2.2 Pause, Stop, Hide
* [x] PASS. Not an issue for RAMP as of August 2016. Care for this criteria will need to be paid if animation / time series support is implemented.

***

# 2.3 Seizures
## 2.3.1 Three Flashes or Below Threshold
* [x] PASS. Not an issue for RAMP as of August 2016. 
***

# 2.4 Navigable
## 2.4.1 Bypass Blocks
### Actions for WCAG "AA" compliance:
* Document missing a "skip to content" link. Not relevant within a WET template. Not sure if this should get addressed in the Full Screen view.

## 2.4.2 Page Titled
### Actions for WCAG "AA" compliance:
* Title contains placeholder text, something more appropriately descriptive will meet this criteria. 
* This is not/should not be an issue when using RAMP in a live environment, however from a demonstrative perspective, we can easily meet this criteria by simply providing some text here. Not relevant within a WET template.

## 2.4.3 Focus Order
TBD
## 2.4.4 Link Purpose (In Context)
TBD
## 2.4.5 Multiple ways
### Actions for WCAG "AA" compliance:
* Site is missing a sitemap. Not sure if this is an achievable goal for this sort of application in full screen. Not relevant within a WET template.

## 2.4.6 Headings and Labels
TBD
## 2.4.7 Focus Visible
TBD

***

# 3.1 Readable
## 3.1.1 Language of Page
TBD
## 3.1.2 Language of Parts
TBD

***

# 3.2 Predictable
## 3.2.1 On Focus
TBD
## 3.2.2 On Input
TBD
## 3.2.3 Consistent Navigation
TBD
## 3.2.4 Consistent Identification
TBD

***

# 3.3 Input Assistance
## 3.3.1 Error Identification
TBD
## 3.3.2 Labels or Instructions
TBD
## 3.3.3 Error Suggestion
TBD
## 3.3.4 Error Prevention (Legal, Financial, Data)
TBD

***

# 4.1 Compatible
## 4.1.1 Parsing
* [x] PASS. Not an issue for RAMP as of August 2016. 
## 4.1.2 Name, Role, Value
* [x] PASS. Not an issue for RAMP as of August 2016. 

***

# General Comments
* Using text-based browsers such as Lynx or Links returns nothing -- is it possible to put some sort of text giving the user an indication that this site requires a modern browser that supports JavaScript, HTML5, etc?
* I have no access to any screen readers presently and cannot advise on this user experience.

**Comment by dan-bowerman** [comment]
[RAMP GIS Viewer BeSD Assessment.xlsx](https://github.com/fgpv-vpgf/fgpv-vpgf/files/450754/RAMP.GIS.Viewer.BeSD.Assessment.xlsx)
